### PR TITLE
JAMES-1717 Vacation should allow user to specify a subject

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java [hunk 1]
```diff
@@ -69,6 +76,9 @@ public VacationReply build() throws MessagingException {
 
         private MimeMessage generateMimeMessage() throws MessagingException {
             MimeMessage reply = (MimeMessage) originalMail.getMessage().reply(NOT_REPLY_TO_ALL);
+            if (subject.isPresent()) {
+                reply.setHeader("subject", subject.get());
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: subject.ifPresent(x -> reply.setHeader("subject", x))

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/VacationReplyTest.java [hunk 2]
```diff
@@ -50,10 +61,30 @@ public void vacationReplyShouldGenerateASuitableAnswer() throws Exception {
 
         assertThat(vacationReply.getRecipients()).containsExactly(originalSender);
         assertThat(vacationReply.getSender()).isEqualTo(originalRecipient);
-        assertThat(vacationReply.getMimeMessage().getHeader("subject")).containsExactly("Re: Original subject");
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/VacationReplyTest.java]: Why did you remove this?

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java (hunk 1 above)
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: setHeader throws MessagingException

This is the more elegant way to deal both with exception handling, and code beauty ;-)

Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: what about Throwing.consumer?

Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: Ah Ok. Pr updated ;-)

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java [hunk 3]
```diff
@@ -76,9 +77,7 @@ public VacationReply build() throws MessagingException {
 
         private MimeMessage generateMimeMessage() throws MessagingException {
             MimeMessage reply = (MimeMessage) originalMail.getMessage().reply(NOT_REPLY_TO_ALL);
-            if (subject.isPresent()) {
-                reply.setHeader("subject", subject.get());
-            }
+            subject.ifPresent((ThrowingConsumer<String>) subjectString -> reply.setHeader("subject", subjectString));
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: In fact it should be Throwing.consumer(subjectString -> reply.setHeader("subject", subjectString)) if I remember correctly.

Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java]: Ok

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:
